torch/autograd/gradcheck.py

```
@@ -206,9 +206,12 @@ def fail_test(msg):
     if any(t.is_sparse for t in tupled_inputs if isinstance(t, torch.Tensor)) and not check_sparse_nnz:
         fail_test('gradcheck expects all tensor inputs '
                   'are dense when check_sparse_nnz is set to False.')
+        return False
```

you should do "return fail_test" rather than separately returning False.